.github/workflows/build.yml

```diff
@@ -1,37 +1,52 @@
 name: Build
 
-on: [push, pull_request]
+on:
+  push:
+    branches-ignore:
+      - 'gh-pages'
+  pull_request:
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - name: "[Setup] Checkout Repository"
-      uses: actions/checkout@v1
-
-    - name: "[Setup] Ensure 'master' branch exists"
-      run: |
-        git fetch --tags
-        git branch --create-reflog master origin/master
-
-    - name: "[Setup] Create GitVersion"
-      uses: docker://gittools/gitversion:5.1.3-linux-netcoreapp2.1
-      with:
-        args: /github/workspace /nofetch /exec /bin/sh /execargs "-c \"echo $GitVersion_FullSemVer > /github/workspace/version.txt\""
-
-    - name: "[Setup] Install .NET Core"
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 3.1.100
-
-    - name: "[Build] NuGet Restore"
-      run: dotnet restore
-
-    - name: "[Build] Build Solution"
-      run: |
-        VERSION=$(cat version.txt)
-        echo $VERSION
-        dotnet build --configuration Release /p:Version=$VERSION;AssemblyVersion=$VERSION
-
-    - name: "[Test] Run Bashi.Console"
-      run: dotnet /home/runner/work/bashi/bashi/src/Bashi.Console/bin/Release/netcoreapp3.1/Bashi.Console.dll --version
+      - name: "[Setup] Checkout Repository"
+        uses: actions/checkout@v1
+
+      - name: "[Setup] Ensure 'master' branch exists"
+        run: |
+          git fetch --tags
+          git branch --create-reflog master origin/master
+
+      - name: "[Setup] Create GitVersion"
+        uses: docker://gittools/gitversion:5.1.3-linux-netcoreapp2.1
+        with:
+          args: /github/workspace /nofetch /exec /bin/sh /execargs "-c \"echo $GitVersion_FullSemVer > /github/workspace/version.txt\""
+
+      - name: "[Setup] Install .NET Core"
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.100
+
+      - name: "[Build] NuGet Restore"
+        run: dotnet restore
+
+      - name: "[Build] Build Solution"
+        run: |
+          VERSION=$(cat version.txt)
+          echo $VERSION
+          dotnet build --configuration Release /p:Version=$VERSION;AssemblyVersion=$VERSION
+
+      - name: "[Test] Run Tests"
+        run: |
+          dotnet test --configuration Release --no-build test/Bashi.Core.Tests/Bashi.Core.Tests.csproj       /p:CollectCoverage=true /p:CoverletOutput=../../coverage/
+          dotnet test --configuration Release --no-build test/Bashi.Console.Tests/Bashi.Console.Tests.csproj /p:CollectCoverage=true /p:CoverletOutput=../../coverage/ /p:MergeWith="../../coverage/coverage.json" /p:CoverletOutputFormat="lcov"
```

Comment:
This should be brought into a simpler command with auto discovery of new test projects.

Reply:
Created #3.